src/components/ebay-table/component.ts

```diff
@@ -1,7 +1,12 @@
+import { AttrTriState } from "marko/tags-html";
 import { WithNormalizedProps } from "../../global";
+import { CheckboxEvent } from "../ebay-checkbox/component-browser";
 
-// export type ColumnType = "normal" | "numeric" | "row-header" | "none";
-export type TableDensity = "compact" | "relaxed" | "none";
+type TableColRowName = string | number;
+export type TableSelectEvent = {
+    selected: Record<TableColRowName, boolean>;
+    allSelected?: AttrTriState;
+};
```

Comment:
Do we need this? Might as well just define this inline. Thats what we do in other places

Reply:
Do you mean `AttrTriState` or this `TableSelectEvent`? [AttrTriState](https://github.com/eBay/ebayui-core/blob/master/src/components/ebay-tri-state-checkbox/component.ts#L23) is the same as tri-state checkbox.

For this TableSelectEvent, I can do it inline. But I find it useful when ppl actually using typescript and need to handle the event.

```MARKO
import { TableSelectEvent } from "<ebay-table>";

class {
  handler(event: TableSelectEvent) {
  // handler
  }
}

<ebay-table on-select("handler") />
```

Reply:
Actually, for event, tri-state checkbox is using same? https://github.com/eBay/ebayui-core/blob/master/src/components/ebay-tri-state-checkbox/component.ts#L4

Reply:
Sure I can remove it. (It won't be inferred tho inside class component)

Reply:
updated please check